relentless/simulate/lammps.py

```diff
@@ -951,28 +955,42 @@ def pair_map(sim, pair):
                         )
                     )
                     fw.write(
-                        "N {N} R {rmin} {rmax}\n\n".format(N=Nr, rmin=r[0], rmax=r[-1])
+                        "N {N} RSQ {rmin} {rmax}\n\n".format(
+                            N=Nr,
+                            rmin=sim.potentials.pair.start,
+                            rmax=sim.potentials.pair.stop,
+                        )
                     )
 
-                    u = sim.potentials.pair.energy((i, j))
-                    f = sim.potentials.pair.force((i, j))
+                    # explicitly use r = sqrt(r^2) to avoid interpolation
+                    u = sim.potentials.pair.energy((i, j), r)
+                    f = sim.potentials.pair.force((i, j), r)
                     for idx in range(Nr):
                         fw.write(
                             "{idx} {r} {u} {f}\n".format(
-                                idx=idx + 1, r=r[idx], u=u[idx], f=f[idx]
+                                idx=idx + 1, r=rsq[idx], u=u[idx], f=f[idx]
                             )
                         )
 
                     # find r where potential and force are zero
-                    nonzero_r = numpy.flatnonzero(
-                        numpy.logical_and(~numpy.isclose(u, 0), ~numpy.isclose(f, 0))
-                    )
-                    if len(nonzero_r) > 1:
-                        # cutoff at last nonzero r (cannot be first r)
-                        rcut[(i, j)] = r[nonzero_r[-1]]
+                    all_rmax = [
+                        pair_pot.coeff[i, j]["rmax"]
+                        for pair_pot in sim.potentials.pair.potentials
+                    ]
+                    if None not in all_rmax:
+                        # use rmax if set for all potentials
+                        rcut[(i, j)] = min(max(all_rmax), sim.potentials.pair.stop)
```

Comment:
Can't take the min of float & product
`TypeError: '<' not supported between instances of 'float' and 'Product'`